source/MaterialXCore/Element.cpp

```diff
@@ -564,7 +564,10 @@ bool ValueElement::validate(string* message) const
         const string& unittype = getUnitType();
         if (!unittype.empty())
         {
-            unitTypeDef = getDocument()->getUnitTypeDef(unittype);
+
+            unitTypeDef = getDocument()->hasDataLibrary() ? 
```

Comment:
Is there a reason this logic isn' just in `Document::getUnitTypeDef()` like the other document query functions?